backend-api/app/routers/system.py

```
@@ -321,32 +290,196 @@ async def run_async(cmd: list[str], timeout: int = 6):
+        matches = re.findall(pattern, html)
+        if not matches:
+            return {"error": "No .deb packages found on Jellyfin repo", "url": JELLYFIN_REPO_BASE}
+        # Sort by version string (last entry = latest)
+        matches.sort(key=lambda m: m[1])
+        latest_filename, latest_version = matches[-1]
+        return {
+            "version": latest_version,
+            "filename": latest_filename,
+            "url": JELLYFIN_REPO_BASE + latest_filename,
+        }
+    except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)